library/bdImage/ControllerHelper/Picker.php

```diff
@@ -40,17 +43,27 @@ public function getPickedData()
             }
         }
 
-        $extraDataInput = new XenForo_Input($this->_controller->getInput()->filterSingle(
-            'bdimage_extra_data',
-            XenForo_Input::ARRAY_SIMPLE
-        ));
+        $extraDataInput = array();
+        if ($prefix === bdImage_ControllerHelper_Picker::EMPTY_PREFIX) {
+            $extraDataInput = array(
+                'is_cover' => $this->_controller->getInput()->filterSingle('is_cover', XenForo_Input::BOOLEAN),
+                'cover_color' => $this->_controller->getInput()->filterSingle('cover_color', XenForo_Input::STRING),
+            );
+        } else {
+            $extraDataInputXen = new XenForo_Input($this->_controller->getInput()->filterSingle(
+                $prefix . 'extra_data',
+                XenForo_Input::ARRAY_SIMPLE
+            ));
+            $extraDataInput = array(
+                'is_cover' => $extraDataInputXen->filterSingle('is_cover', XenForo_Input::BOOLEAN),
+                'cover_color' => $extraDataInputXen->filterSingle('cover_color', XenForo_Input::STRING),
+            );
+        }
```

Comment:
Rút gọn lại nha chế.

```suggestion
        $input = $this->_controller->getInput();
        if ($prefix !== bdImage_ControllerHelper_Picker::EMPTY_PREFIX) {
            $input = new XenForo_Input($input->filterSingle(
                $prefix . 'extra_data',
                XenForo_Input::ARRAY_SIMPLE
            ));
        }
        $extraDataInput = $input->filter(array(
            'is_cover' => XenForo_Input::BOOLEAN,
            'cover_color' => XenForo_Input::STRING
        ));
```